autoload/ale/handlers/solhint.vim

```diff
@@ -18,29 +18,30 @@ function! ale#handlers#solhint#Handle(buffer, lines) abort
     " /path/to/file/file.sol: line 1, col 10, Error - 'addOne' is defined but never used. (no-unused-vars)
     let l:output = []
 
-    let l:lint_pattern = '\v^[^:]+: line (\d+), col (\d+), (Error|Warning) - (.*) \((.*)\)$'
+    " let l:lint_pattern = '\v^[^:]+: line (\d+), col (\d+), (Error|Warning) - (.*) \((.*)\)$'
+    let l:lint_pattern = '\v^[^:]+:(\d+):(\d+): %(Parse error: )@<!\ze(.*)\s+\[(Error|Warning)\/([^\]]+)\]$'
 
     for l:match in ale#util#GetMatches(a:lines, l:lint_pattern)
-        let l:isError = l:match[3] is? 'error'
+        let l:isError = l:match[4] is? 'error'
         call add(l:output, {
         \   'lnum': l:match[1] + 0,
         \   'col': l:match[2] + 0,
-        \   'text': l:match[4],
+        \   'text': l:match[3],
         \   'code': l:match[5],
         \   'type': l:isError ? 'E' : 'W',
         \})
     endfor
 
-    let l:syntax_pattern = '\v^[^:]+: line (\d+), col (\d+), (Error|Warning) - (Parse error): (.*)$'
+    " let l:syntax_pattern = '\v^[^:]+: line (\d+), col (\d+), (Error|Warning) - (Parse error): (.*)$'
```

Comment:
I understand solhint no longer supports the older formatter but this change is too recent and some users may be still stuck using older versions and be required to use the current formatter.  In this cases this would be a breaking change on ALE's side.  Instead of replacing the previous pattern is better to support both.

Simply keep the previous code intact and add a second pass over the lines using the new pattern adding the matches to the output array.

Reply:
Hey... Sorry about the long delay.
Apparently `solhint` added the `compact` formatter back after I made this PR.
However, the `unix` formatter is available in every single version, so these changes would not break the integration.

Apparently, only `v3.4.0` is missing the `compact` format though, so it might be the case that this PR is not required anymore.